lib/commons/color/get-background-color.js

```diff
@@ -65,9 +77,48 @@ function calculateObscuringAlpha(elmIndex, elmStack) {
 }
 
 function elmOutsideParent(elm, bgElm, bgColor) {
```

Comment:
I'll look over these functions to see they can be refactored to be smaller and still cover the necessary use cases. But speaking of those, @dylanb do you remember what the Capital One bug was you mention in this commit? That's where `bgColor.alpha` originated, so I want to make sure it doesn't regress. https://github.com/dequelabs/axe-core/commit/6557ff85f9a9a523f831c0642a20b335746d00cd